File: BAEKJOON/[1940] 주몽.java
import java.util.*;

public class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		int m = sc.nextInt();
		int []material = new int[n];
		for (int i = 0; i < n; i++) {
			material[i] = sc.nextInt();
		}
		Arrays.sort(material);
		
		int answer = 0;
		int start = 0;
		int end = n - 1;
		int sum = 0;
		while (start < end) {
			sum = material[start] + material[end];
			if (sum == m) {
				answer++;
				start++;
				end--;
			}
			else if(sum < m) {
				start++;
			}
			else {
				end--;
			}
		}
		System.out.print(answer);
	}
}

File: BAEKJOON/[1940] 주몽.py
n = int(input())
m = int(input())
material = list(map(int, input().split()))

start = 0
end = n - 1
answer = 0

material.sort()

while(start < end) :
  sum = material[start] + material[end]
  if sum == m :
    answer += 1
    start += 1
    end -= 1
  elif sum > m :
    end -= 1
  else :
    start += 1

print(answer)
